WordPressKit/WordPressComRestApi.swift

```diff
@@ -592,3 +687,28 @@ private extension WordPressComRestApi {
 extension ProgressUserInfoKey {
     public static let sessionTaskKey = ProgressUserInfoKey(rawValue: WordPressComRestApi.SessionTaskKey)
 }
+
+// MARK: - POST encoding
+
+private extension Dictionary {
+    func percentEncoded() -> Data? {
+        return map { key, value in
+            let escapedKey = "\(key)".addingPercentEncoding(withAllowedCharacters: .urlQueryValueAllowed) ?? ""
+            let escapedValue = "\(value)".addingPercentEncoding(withAllowedCharacters: .urlQueryValueAllowed) ?? ""
+            return escapedKey + "=" + escapedValue
+        }
```

Comment:
Nitpick. In the case where either percent encoding fails, we may end up with something inconsistent (e.g. a `=` by itself, or `foo=` or `=bar`.

What do you think of this version that uses `compactMap` to filter out those occurrences?

```suggestion
return compactMap { key, value in
            guard
                let escapedKey = "\(key)".addingPercentEncoding(withAllowedCharacters: .urlQueryValueAllowed),
                let escapedValue = "\(value)".addingPercentEncoding(withAllowedCharacters: .urlQueryValueAllowed)
            else {
                return .none
            }

            return escapedKey + "=" + escapedValue
        }
```

It compiles and the tests pass locally for me, but I don't see an explicit test for different behavior of this new method, so there could be something I lost in translation.